src/checkers/inference/model/TernaryVariableSlot.java

```diff
@@ -0,0 +1,90 @@
+package checkers.inference.model;
+
+/**
+ * TernaryVariableSlots represent locations whose values depend on two other VariableSlots.
```

Comment:
From this line, I think you are saying `TernaryVariableSlot` is common super type for `VPResultSlot` and `LeastUpperBoundSlot`.

Reply:
Just like we have `BinaryConstraint` but not `BinaryVariableSlot`, I would prefer no `TernaryVariableSlot`. As `TernaryXXX` is already very general way of saying slots(sth. related to three slots), having both `TernaryVariableSlot` and `TernaryConstraint` seems very confusing for me as a user.